### ISSUE TITLE
Body is expected to be empty when httpStatus is 404

### ISSUE THREAD

**Issue by rcaval** [DiUS/pact-jvm#122]
We have a scenario where a provider, when responding with a 404 it also sends a response body.
What we noticed though is that pactVerify expect the body to be empty when the http response status is 404.
Example  pact.json file
```json
{
  "provider": {
    "name": "Animal_Service"
  },
  "consumer": {
    "name": "Zoo_App"
  },
  "interactions": [
    
    {
      "provider_state": "there is not an alligator named Mary",
      "description": "a request for alligator Mary",
      "request": {
        "method": "get",
        "path": "/alligators/Mary"
      },
      "response": {
        "status": 404,
        "body": {
          "errorMessage": "Resource not found!"
        }
      }
    }
  ],
  "metadata": {
    "pact_gem": {
      "version": "1.0.9"
    }
  }
}

```
console output:

```
Given there is not an alligator named Mary
  a request for alligator Mary
    returns a response which
      has status code 404 (OK)
      has a matching body (FAILED)


Failures:

0) Verifying a pact between myConsumer and myProvider - a request for alligator Mary Given there is not an alligator named Mary returns a response which has a matching body
      $.body -> Expected errorMessage='Resource not found!' but was missing

      Diff:

      @1
      -    "errorMessage": "Resource not found!"
      +
      }
```

**Comment by uglyog** [comment]
In this case, pactVerify is expecting a body to be returned, but it did not receive one. Can you verify that your provider is sending a body?

Have a look at issue #115, which was fixed by upgrading the version of the pact-jvm library.

**Comment by tmssoares** [comment]
Hey @uglyog , we ran the following interaction:
```
 {
      "description": "error scenario: expect a 404",
      "request": {
        "method": "GET",
        "path": "http://localhost:8070/item-service/item/00000000-0000-0000-0000-000000000000"
      },
      "response": {
        "status": 404,
        "headers": {
          "Content-Type": "application/json;charset=UTF-8"
        },
        "body": {
        }
      }
    }
```
We had the following error message:
```
  error scenario: expect a 404
    returns a response which
      has status code 404 (OK)
      includes headers
        "Content-Type" with value "application/json;charset=UTF-8" (OK)
      has a matching body (OK)
:pactVerify

BUILD SUCCESSFUL
```
And it is fine. The problem occurs when we try to test the response body, that contains an error message as described following:
```
 
{
      "description": "error scenario: expect a 404",
      "request": {
        "method": "GET",
        "path": "http://localhost:8070/item-service/item/00000000-0000-0000-0000-000000000000"
      },
      "response": {
        "status": 404,
        "headers": {
          "Content-Type": "application/json;charset=UTF-8"
        },
        "body": {
          "errorMessage": "Item not found!"
        }
      }
    }
```
We see the following message:
```
error scenario: expect a 404
    returns a response which
      has status code 404 (OK)
      includes headers
        "Content-Type" with value "application/json;charset=UTF-8" (OK)
      has a matching body (FAILED)

Failures:

0) Verifying a pact between providerService and consumerService - error scenario: expect a 404 returns a response which has a matching body
      $.body -> Expected errorMessage='Item not found!' but was missing

      Diff:

      @1
      -    "errorMessage": "Item not found!"
      +
      }
```
Accessing using the browser I can see the message:
```
{
"ts": "2015-05-25T14:15:28.671Z",
"httpStatus": 404,
"errorMessage": "Item not found"
}
```

Thanks!!

**Comment by uglyog** [comment]
What version of the pact library are you using?

**Comment by tmssoares** [comment]
This is the version: 
```
plugins {
    id "au.com.dius.pact" version "2.1.1"
}
```

The gradle version is 2.1.

**Comment by rcaval** [comment]
It is fixed in version 2.2.1. Thanks.

This works:
```
plugins {
    id "au.com.dius.pact" version "2.2.1"
}
```
